packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -4497,66 +4497,452 @@ class PositionedDirectional extends StatelessWidget {
+///    axis, and the child is trying to be the max possible size in that
+///    axis, then the error you will see is likely _"BoxConstraints forces an
+///    infinite (height or width)"_.
+///
+/// 2. Children that are not [Flexible]s or [Expanded]s have a fixed size
+///    and so do not have a flex parameter. They will not go through steps
+///    2 or 3.
+///
+///    In the main axis, take the space left over and divide it among the
+///    [Flexible]s. (Again, [Expanded] is just a kind of [Flexible]).
+///
+///    The [Flexible.flex] parameter determines how much of the
+///    leftover space each [Flexible] will have reserved
+///    for it. Notice _"reserved for it"_ does not mean how big
+///    it's going to be. It means the space (or size) reserved for it
+///    is the maximum size the [Flexible] _could be_, in the main axis.
+///
+///    Add up all the [flex] values of each [Flexible]'s fit parameter,
+///    and each [Flexible] will have a percentage of the leftover space
+///    reserved for it that will be equal to the percentage
+///    of total [flex] each [Flexible] has. For example:
+///
+///    [Flexible] 1 has [flex] of 50.
+///    [Expanded] 1 has [flex] of 30.
+///    [Expanded] 2 has [flex] of 20.
+///
+///    In the above case, the total of all flex parameters is 100. The
+///    [Flexible] will get 50/100 (50%) of the remaining space reserved
+///    for it. The first [Expanded] will get 30/100 (30%) reserved for it,
+///    and the second one will get 20% of the remaining space
+///    reserved for it.
+///
+///    The numbers you use for your [Flexible.fit] parameters don't have to
+///    add to 100, the process remains the same. Making them total 100
+///    is simply easier to visualize.
+///
+/// 3. _Keep in mind that all of step 3 applies only to the main axis._
+///    Nothing in Step 3 affects the size of the children in the cross axis.
+///
+///    Lay out the [Flexible] and [Expanded] children, and use
+///    the same cross axis size limits (constraints) that were used
+///    in Step 1. If using [CrossAxisAlignment.stretch] then
+///    force all of the children to be the maximum possible size in the cross
+///    axis, otherwise let them be the size they want.
+///
+///    Unlike in Step 1, this time set _maximum_ main axis constraints
+///    (max size limits) for each [Flexible] and [Expanded] to be the
+///    sizes that were reserved for them in Step 2. In other words, the
+///    maximum possible main axis sizes were calculated in Step 2, and now
+///    those sizes get used when laying out each [Flexible] and [Expanded].
+///
+///    An [Expanded] will pass its child tight constraints, a
+///    [Flexible] will pass its child loose constraints (if the fit
+///    parameter hasn't been changed from its default). Children of an
+///    [Expanded] (or a [Flexible] with it's fit property set to
+///    [FlexFit.tight]) are given tight constraints.
```

Comment:
This last sentence seems somewhat redundant with the first. Maybe just describe tight constraints in one sentence, and loose constraints in the second sentence?

Reply:
Handling this offline in the IDE.